packages/front/svelte/components/SubventionsVersementsDashboard/SubventionTable/SubventionTable.controller.js

```diff
@@ -67,23 +109,15 @@ export default class SubventionTableController {
     updateElements(elements) {
         this.elements = elements;
 
-        const sizedTrim = value => trim(value, this.MAX_CHAR_SIZE);
         const elementsDataViews = this.elements.map(element => {
             if (!element.subvention) return null;
 
-            const projectName = valueOrHyphen(this.getProjectName(element.subvention));
-
             return {
+                ...SubventionTableController._extractTableDataFromElement(element),
                 subvention: element.subvention,
-                serviceInstructeur: sizedTrim(element.subvention.service_instructeur),
-                dispositif: valueOrHyphen(sizedTrim(element.subvention.dispositif)),
-                projectName: projectName,
-                projectNamePosition: projectName === "-" ? "center" : "start",
-                enableButtonMoreInfo: !!(element.subvention.actions_proposee?.length || 0),
-                montantsDemande: valueOrHyphen(numberToEuro(element.subvention.montants?.demande)),
-                montantsAccordeOrStatus: element.subvention.montants?.accorde
-                    ? numberToEuro(element.subvention.montants?.accorde)
-                    : element.subvention.status
+                // this.projectName comes from _extractTableDataFromElement()
```

Comment:
comprends pas ? this représente ta class subvention table pas l'objet que tu est entrain de créer

Reply:
Oui t'as raison ça doit pas du tout fonctionner. Je modifie ça